### PR TITLE
[FLINK-18607][build] Give the maven module a human readable name

### PR DESCRIPTION
**Pull request by nielsbasjes**
## What is the purpose of the change

Make the build output more human readable for the developers van various situations.

## Brief change log

Add or change the `<name>` of each maven module (i.e. in every pom.xml)

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

The simplest way to test is by doing a `mvn clean` and verify the list of names at the end looks good.
Also looking at the change set verify that the names for each module are correct.

In some exceptional cases I made the name reflect a bit more the purpose like in the case of 
`flink-examples/flink-examples-build-helper` where the description indicates `This is a utility module for building example jars to be used in flink-dist.` so I named it `Flink : Examples : Dist : `

Fun fact: I noticed that the ordering of the modules during the build is changed a lot by maven because apparently the ordering of the modules in the pom.xml does not meet the configured dependencies between the modules. So maven reorders the build by itself.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): **no**
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: **no**
  - The serializers: **no**
  - The runtime per-record code paths (performance sensitive): **no**
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: **no**
  - The S3 file system connector: **no**

## Documentation

  - Does this pull request introduce a new feature? **no**

### CODE REVIEW

**Comment by nielsbasjes** [comment]
Note that all '<packaging>pom</packaging>'  modules have a name that ends with `<space>:<space>` because that makes the ordering in the Maven overview in IntelliJ better.

Also note that the naming convention I show here is based upon my own preference: what I find easy to read.

File: flink-examples/flink-examples-build-helper/flink-examples-streaming-gcp-pubsub/pom.xml [hunk 1]
```diff
@@ -29,7 +29,7 @@ under the License.
 	</parent>
 
 	<artifactId>flink-examples-streaming-gcp-pubsub_${scala.binary.version}</artifactId>
-	<name>flink-examples-streaming-gcp-pubsub</name>
+	<name>Flink : Examples : Dist : Streaming Google PubSub</name>
```
Comment (KurtYoung) [inline flink-examples/flink-examples-build-helper/flink-examples-streaming-gcp-pubsub/pom.xml:32]: Dist -> Helper?

Comment (nielsbasjes) [reply flink-examples/flink-examples-build-helper/flink-examples-streaming-gcp-pubsub/pom.xml:32]: Yes, can be. I had doubts about this one. The artifact name is build helper yet the purpose is making it available in the dist.

Comment (nielsbasjes) [reply flink-examples/flink-examples-build-helper/flink-examples-streaming-gcp-pubsub/pom.xml]: For now I made it consistent with the module artifact name : "Build Helper".

Comment (nielsbasjes) [reply flink-examples/flink-examples-build-helper/flink-examples-streaming-gcp-pubsub/pom.xml]: I personally would rename this to something like "dist helper" or "dist examples" but that is a separate change if desired.

**Comment by KurtYoung** [comment]
@nielsbasjes Thanks for the update, I will merge this once tests passed.

**Pull request merged**